File: src/main/java/org/example/collection/MusicGenre.java
package org.example.collection;

import java.util.LinkedList;
/**
 * Класс перечисление жанров музыки
 */
public enum MusicGenre{
    JAZZ,
    SOUL,
    POP;
    /**
     * @return возвращает все значения перечисления
     */
    public static String getValue() {
        LinkedList<String> enumValues = new LinkedList<>();
        for (MusicGenre genre :  MusicGenre.values()){
            enumValues.add(genre.toString());
        }
        return String.join(", ", enumValues);
    }
}

File: common/src/main/kotlin/collection/MusicGenre.kt
package collection

import java.io.Serializable
import java.util.*
import java.util.stream.Collectors


enum class MusicGenre: Serializable {
    JAZZ,
    SOUL,
    POP;

    companion object {
        val value: String

            get() = Arrays.stream(entries.toTypedArray())
                .map { obj: MusicGenre -> obj.name }
                .collect(Collectors.joining(", "))
    }
}